### DISCUSSION TITLE
Connection to SQLite or JSON file

### DISCUSSION THREAD

**Discussion by ngomory** [capcom6/android-sms-gateway#123] [Ideas]
Hi

First of all, thanks for making this project. It's really great and useful.

I have a proposal idea at the database level in the configuration file: config.yml
Add a connection to a SQLite or JSON file so that you can quickly start the Docker image in the same folder in external dependency.

Thanks

**Comment by capcom6** [comment]
Hello!

Thank you for your proposal.

Could you please provide more details on why you prefer SQLite or JSON over MySQL/MariaDB? If you're already using Docker, starting a database container is quite straightforward. For reference, you can check Single File Deployment guide here: https://github.com/capcom6/sms-gateway/tree/master/deployments/docker-compose-proxy

If you'd prefer not to share this information publicly, feel free to contact me via email at support@sms-gate.app.

**Comment by ngomory** [comment]
Hello

Thanks for your answer.
Indeed, I started the Docker image, but the application cannot connect to the database located on the same server which is perfectly functional and accessible.
But the application can connect to another database located on another server.

# docker compose
services:
  gateway:
    image: capcom6/sms-gateway:latest
    volumes:
      - ./gateway/config.yml:/app/config.yml
    ports:
      - "9005:3000"

# config.yml 
gateway: 
  mode: private 
  private_token: xxxxxxxxx
http: 
  listen: 0.0.0.0:3000 
database: 
  dialect: mysql 
  host: xxx.xxx.xxx.xxx
  port: 3306
  user: xxxxxxxxx 
  password: xxxxxxxxx
  database: xxxxxxxxx
  timezone: UTC

# Here is the error I have in the console:

{"level":"error","ts":1726933822.9150367,"caller":"fxevent/zap.go:59","msg":"invoke failed","error":"dial tcp xxx.xxx.xxx.xxx:3306: connect: connection timed out","stack":"github.com/capcom6/go-infra-fx/cli.GetModule\n\t/go/pkg/mod/github.com/capcom6/go-infra-fx@v0.0.2/cli/module.go:30\ngithub.com/capcom6/sms-gateway/internal/sms-gateway.Run\n\t/go/src/internal/sms-gateway/app.go:51\nmain.main\n\t/go/src/cmd/sms-gateway/main.go:28\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271\n","function":"github.com/capcom6/go-infra-fx/db.Migrate()","module":"cli","stacktrace":"go.uber.org/fx/fxevent.(*ZapLogger).logError\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/fxevent/zap.go:59\ngo.uber.org/fx/fxevent.(*ZapLogger).LogEvent\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/fxevent/zap.go:193\ngo.uber.org/fx.(*module).executeInvoke\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/module.go:289\ngo.uber.org/fx.(*module).executeInvokes\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/module.go:274\ngo.uber.org/fx.(*module).executeInvokes\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/module.go:268\ngo.uber.org/fx.New\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/app.go:503\ngithub.com/capcom6/sms-gateway/internal/sms-gateway.Run\n\t/go/src/internal/sms-gateway/app.go:50\nmain.main\n\t/go/src/cmd/sms-gateway/main.go:28\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}

**Reply by capcom6** [reply]
This is not an application-related error, it's a network issue.

Because I don't know your network infrastructure, I can't provide specific hints without additional information. You mention that the database is located on the same server, but is it running in Docker or directly on the host machine? In either case, you'll need to make some changes to your `docker-compose` file.

To access a resource in another container, both containers must be in the same Docker network. To access the host machine, you can use the special address `host.docker.internal`, as described in the Docker documentation: https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host

For more information on Docker networking, you can refer to the official Docker documentation: https://docs.docker.com/engine/network/

**Comment by ngomory** [comment]
Hi

I also did the test with this docker compose without modifying anything and I still have the same error: 

=> https://github.com/capcom6/sms-gateway/blob/master/deployments/docker-compose/docker-compose.yml

Erreur : 

{"level":"error","ts":1729018765.5882807,"caller":"fxevent/zap.go:59","msg":"start failed","error":"dial tcp [::1]:3306: connect: connection refused","stacktrace":"go.uber.org/fx/fxevent.(*ZapLogger).logError\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/fxevent/zap.go:59\ngo.uber.org/fx/fxevent.(*ZapLogger).LogEvent\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/fxevent/zap.go:215\ngo.uber.org/fx.(*App).Start.func1\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/app.go:640\ngo.uber.org/fx.(*App).Start\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/app.go:645\ngo.uber.org/fx.(*App).run\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/app.go:587\ngo.uber.org/fx.(*App).Run\n\t/go/pkg/mod/go.uber.org/fx@v1.20.1/app.go:578\ngithub.com/capcom6/sms-gateway/internal/sms-gateway.Run\n\t/go/src/internal/sms-gateway/app.go:58\nmain.main\n\t/go/src/cmd/sms-gateway/main.go:27\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}

Best regards

**Reply by capcom6** [reply]
Hello!

It seems like something has changed, because in the log, the server tries to connect to the database at `localhost`. However, in the original `docker-compose.yml`, the `DATABASE__HOST` environment variable points to the `db` host in the same `docker-compose` file.

I've just tested the same `docker-compose` file by running the `make docker` command at the root of the project, and everything works well.

To troubleshoot this issue, you could try the following steps:

1. Ensure you're using the latest version of the `docker-compose.yml` file.
2. Double-check that you haven't accidentally modified any environment variables.
3. Try removing all containers and volumes related to this project, then rebuild and restart.

If the problem persists, please provide more details about your environment and any changes you might have made to the configuration.
